infrastructure/task/scripts/lagoon-get-backup.sh

```diff
@@ -0,0 +1,30 @@
+#!/usr/bin/env bash
+# Restore and download files from a Lagoon backup.
+set -eo pipefail
+
+LAGOON_ENVIRONMENT="${LAGOON_ENVIRONMENT:-main}"
+# The entry in the list of backups for the provided type to retrieve for the
+# given backup type. 1 is the most recent.
+BACKUP_ENTRY_INDEX=$((${BACKUP_ENTRY:-1}-1));
+
+if [[ -z "${LAGOON_PROJECT}" || -z "${BACKUP_TYPE}" || -z "${BACKUP_DESTINATION}" ]]; then
+	echo "usage: LAGOON_PROJECT=<LAGOON_PROJECT> BACKUP_TYPE=<BACKUP_TYPE> BACKUP_DESTINATION=<BACKUP_DESTINATION> $0" >&2
+	exit 1
+fi
+
+BACKUPS=$(lagoon list backups -p "${LAGOON_PROJECT}" -e "${LAGOON_ENVIRONMENT}" --output-json);
+BACKUP_ID=$(echo "$BACKUPS" | jq -r ".data | map(select(.source == \"${BACKUP_TYPE}\")) | nth(${BACKUP_ENTRY_INDEX}) | .backupid");
+BACKUP_RESULT="Error";
+echo -e "\nRetrieving ${BACKUP_ENTRY}. ${BACKUP_TYPE} backup with id ${BACKUP_ID} from ${LAGOON_ENVIRONMENT} environment of ${LAGOON_PROJECT} project \n\n";
+# Wait a while we wait for the backup to become available.
+# It must be retrieved before it can be downloaded.
+while [[ $BACKUP_RESULT == "Error"* ]]; do
```

Comment:
This will loop forever if the lagoon keeps returning error?

And it's painfully obvious that their CLI is a thin wrapper around their API. There ought to be a single command to `retrive`, `get` and download the backup. These commands is obviously implemented to check checkboxes, not to solve a use case.

Reply:
> This will loop forever if the lagoon keeps returning error?

In essence: Yes 😕 .